pkg/object/ceph.go

```diff
@@ -230,24 +230,76 @@ func (c *ceph) ListAll(prefix, marker string, followLink bool) (<-chan Object, e
 	}
 	// the keys are not ordered, sort them first
 	sort.Strings(keys)
+	c.release(ctx)
 
 	var objs = make(chan Object, 1000)
-	// TODO: parallel
+	var concurrent = 20
+	ms := make([]sync.Mutex, concurrent)
+	conds := make([]*sync.Cond, concurrent)
+	ready := make([]bool, concurrent)
+	results := make([]Object, concurrent)
+	errs := make([]error, concurrent)
+	for j := 0; j < concurrent; j++ {
+		conds[j] = sync.NewCond(&ms[j])
+		if j < len(keys) {
+			go func(j int) {
+				ctx, err := c.newContext()
+				if err != nil {
+					logger.Errorf("new context: %s", err)
+					errs[j] = err
+					return
+				}
+				defer ctx.Destroy()
```

Comment:
use `c.release`?

Reply:
we release the context when there is on error